operator/controllers/reconcilers/seldon/runtime_reconciler.go

```diff
@@ -27,9 +31,88 @@ type SeldonRuntimeReconciler struct {
 	configMapReconciler  common.Reconciler
 }
 
+func ParseInt32(s string) (int32, error) {
+	i64, err := strconv.ParseInt(s, 10, 32)
+	return int32(i64), err
+}
+
+func ValidateDataflowScaleSpec(
+	component *mlopsv1alpha1.ComponentDefn,
+	runtime *mlopsv1alpha1.SeldonRuntime,
+	commonConfig common.ReconcilerConfig,
+	namespace *string,
+) error {
+	ctx, clt, recorder := commonConfig.Ctx, commonConfig.Client, commonConfig.Recorder
+	numPartitions, err := ParseInt32(runtime.Spec.Config.KafkaConfig.Topics["numPartitions"].StrVal)
+	if err != nil {
+		return fmt.Errorf("failed to parse numPartitions from KafkaConfig: %w", err)
+	}
+
+	var pipelineCount int32 = 0
+	if namespace != nil {
+		// Get the number of Pipeline resources in the namespace
+		var pipelineList mlopsv1alpha1.PipelineList
+		if err := clt.List(ctx, &pipelineList, client.InNamespace(*namespace)); err != nil {
+			return fmt.Errorf("failed to list Pipeline resources in namespace %s: %w", *namespace, err)
+		}
+		pipelineCount = int32(len(pipelineList.Items))
+	}
+
+	maxReplicas := numPartitions
+	if pipelineCount != 0 {
+		maxReplicas = numPartitions * pipelineCount
+	}
+
+	if component.Replicas != nil && *component.Replicas > maxReplicas {
+		component.Replicas = &maxReplicas
+		recorder.Eventf(
+			runtime,
+			v1.EventTypeWarning,
+			"DataflowEngineReplicasAdjusted",
+			fmt.Sprintf("Dataflow engine replicas adjusted to %d based on KafkaConfig and Pipeline count", maxReplicas),
```

Comment:
Might be worth being slightly more explicit in the log message about where the KafkaConfig lies in an installed cluster. It is in `SeldonConfig`, (overwrite-able by `SeldonRuntime`) right?